apicast/src/provider.lua

```diff
@@ -372,16 +376,25 @@ function _M.access(service)
     params.app_id = parameters[credentials.app_id]
     params.app_key = parameters[credentials.app_key] -- or ""  -- Uncoment the first part if you want to allow not passing app_key
 
+    -- XXX: why do we need to include the app_key into the cached key? the app (and its access) are identified
+    -- just by the app_id
     ngx.var.cached_key = concat({service.id, params.app_id, params.app_key}, ':')
 
   elseif backend_version == 'oauth' then
+    -- XXX: for OAuth the access_token is usually passed in the Authorization header:
+    -- Authorization: Bearer <access_token> 
+    -- we need to cover this
+    -- XXX: we should be able to build $credentials (access_token=<access_token) the same way as for 
```

Comment:
`ngx.var.access_token` was actually only used in one location, which was never used anyway.
PR for removing it: https://github.com/3scale/apicast/pull/164